Tests/Marketplace/search_and_install_packs.py

```diff
@@ -168,8 +169,98 @@ def search_pack(client: demisto_client, prints_manager: ParallelPrintsManager, p
         lock.release()
 
 
+def find_malformed_pack_id(error_message: str):
+    """
+    Find the pack ID from the installation error message.
+    Args:
+        error_message (str): The error message of the failed installation pack.
+
+    Returns: Pack_id (str)
+
+    """
+    malformed_pack_pattern = re.compile(r'invalid version [0-9.]+ for pack with ID ([\w_-]+)')
+    malformed_pack_id = malformed_pack_pattern.findall(str(error_message))
+    if malformed_pack_id:
+        return malformed_pack_id
+    else:
+        raise Exception(f'The request to install packs has failed. Reason: {str(error_message)}')
```

Comment:
Raising exception here will not break the while loop.
You should handle the case where you don't find a pack outside this method and break the loop